libs/server/src/lib/api/read-recent-actions.ts

```diff
@@ -0,0 +1,34 @@
+import { Store } from '@nrwl/angular-console-enterprise-electron';
+import { RecentAction } from '../generated/graphql-types';
+
+function getRecentActionsKey(projectPath: string): string {
+  return `recentActions:${projectPath}`;
+}
+
+export function readRecentActions(
+  store: Store,
+  projectPath: string
+): RecentAction[] {
+  const actions: any[] = store.get(getRecentActionsKey(projectPath));
+  return (actions || []).filter(action => action && !!action.actionName);
+}
+
+export function storeTriggeredAction(
+  store: Store,
+  projectPath: string,
+  actionName: string,
+  schematicName?: string
+) {
+  const MAX_RECENT_ACTIONS = 5;
+  const existingActions = readRecentActions(store, projectPath);
+  store.set(getRecentActionsKey(projectPath), [
+    { actionName, schematicName },
+    ...existingActions
+      .filter(
+        action =>
+          action.actionName !== actionName ||
+          action.schematicName !== schematicName
+      )
+      .slice(0, MAX_RECENT_ACTIONS - 1)
```

Comment:
You don't need the - 1 here since the upper bound of slice is not included

Reply:
I need the - 1 because I'm adding `{ actionName, schematicName }` at the front and want to remove the last element (assuming we've reached the `MAX_RECENT_ACTIONS`)

Reply:
I'll move the slice outside the `]` and take off the `- 1` so that I don't confuse future readers